### PR TITLE
hyperv: update support to features exposed in libvirt >= 4.10

### PR DESCRIPTION
**Pull request by fromanirh**
**What this PR does / why we need it**:
Update hyperv support to libvirt 4.10 and onwards. Since we ship libvirt 5.1 in virt-launcher containers, we can use all the new hyperv features

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubevirt/kubevirt/issues/1919

**Special notes for your reviewer**:
Requires labels as provided by https://github.com/fromanirh/kvm-info-nfd-plugin >= 0.4.0
It will be handled by nfd configuration

**Release note**:
```release-note
Add support for hyperv enlightenments provided by libvirt >= 4.10
```

### CODE REVIEW

File: api/openapi-spec/swagger.json [hunk 1]
```diff
@@ -4463,12 +4463,28 @@
    "v1.FeatureHyperv": {
     "description": "Hyperv specific features.",
     "properties": {
+     "evmcs": {
+      "description": "EVMCS Speeds up L2 vmexits, but disables other virtualization features. Requires vapic\nDefaults to the machine type setting.\n+optional",
```
Comment (phoracek) [inline api/openapi-spec/swagger.json]: Requires vapic**.**

File: pkg/api/v1/schema.go [hunk 2]
```diff
@@ -789,14 +789,34 @@ type FeatureHyperv struct {
...
-	// Reset enables Hyperv reboot/reset for the vmi.
+	// Reset enables Hyperv reboot/reset for the vmi. Requires synic.
 	// Defaults to the machine type setting.
 	// +optional
 	Reset *FeatureState `json:"reset,omitempty"`
 	// VendorID allows setting the hypervisor vendor id.
 	// Defaults to the machine type setting.
 	// +optional
 	VendorID *FeatureVendorID `json:"vendorid,omitempty"`
+	// Frequencies improve Hyper-V on KVM (TSC clock source)
```
Comment (phoracek) [inline pkg/api/v1/schema.go]: source)**.**

File: pkg/api/v1/schema.go [hunk 3]
```diff
@@ -789,14 +789,34 @@ type FeatureHyperv struct {
...
+	// Defaults to the machine type setting.
+	// +optional
+	Frequencies *FeatureState `json:"frequencies,omitempty"`
+	// Reenlightenment improve Hyper-V on KVM (TSC clock source)
```
Comment (phoracek) [inline pkg/api/v1/schema.go]: ditto

File: pkg/api/v1/schema.go [hunk 4]
```diff
@@ -789,14 +789,34 @@ type FeatureHyperv struct {
...
+	Reenlightenment *FeatureState `json:"reenlightenment,omitempty"`
+	// TLBFlush improves performances in overcommited environments. Requires vpindex.
+	// Defaults to the machine type setting.
+	// +optional
+	TLBFlush *FeatureState `json:"tlbflush,omitempty"`
+	// IPI improves performances in overcommited environments. Requires vpindex.
+	// Defaults to the machine type setting.
+	// +optional
+	IPI *FeatureState `json:"ipi,omitempty"`
+	// EVMCS Speeds up L2 vmexits, but disables other virtualization features. Requires vapic
```
Comment (phoracek) [inline pkg/api/v1/schema.go]: vapic**.**

File: api/openapi-spec/swagger.json (hunk 1 above)
Comment (phoracek) [inline api/openapi-spec/swagger.json]: Default from machine type setting are given automatically by libvirt?

Comment (fromanirh) [reply api/openapi-spec/swagger.json]: Done

File: pkg/api/v1/schema.go (hunk 2 above)
Comment (fromanirh) [reply pkg/api/v1/schema.go]: Done.

File: pkg/api/v1/schema.go (hunk 3 above)
Comment (fromanirh) [reply pkg/api/v1/schema.go]: done

File: pkg/api/v1/schema.go (hunk 4 above)
Comment (fromanirh) [reply pkg/api/v1/schema.go]: done

File: api/openapi-spec/swagger.json (hunk 1 above)
Comment (fromanirh) [reply api/openapi-spec/swagger.json]: Uhm, here I just (mindlessly) copy-pasted the same sentence from all the existing features. I'm not sure this is accurate. Will address this in another PR, I'll keep the sentence for this one.

Comment (phoracek) [reply api/openapi-spec/swagger.json]: I see. Right, I was just curious

File: pkg/virt-controller/services/template.go [hunk 5]
```diff
@@ -200,26 +205,60 @@ func getHypervNodeSelectors(vmi *v1.VirtualMachineInstance) (map[string]string,
...
-	if isFeatureStateEnabled(hyperv.Runtime) {
-		nodeSelectors[NFD_KVM_INFO_PREFIX+"runtime"] = "true"
-	}
-	if isFeatureStateEnabled(hyperv.Reset) {
-		nodeSelectors[NFD_KVM_INFO_PREFIX+"reset"] = "true"
-	}
-	if isFeatureStateEnabled(hyperv.SyNIC) {
-		nodeSelectors[NFD_KVM_INFO_PREFIX+"synic"] = "true"
-		// TODO: SyNIC depends on vp-index on QEMU level. We should enforce this constraint.
+	hvFeatureLabels := []hvFeatureLabel{
```
Comment (davidvossel) [inline pkg/virt-controller/services/template.go]: What would you think about moving this array out of this function and defining it as a var in the package?

Comment (fromanirh) [reply pkg/virt-controller/services/template.go]: fine for me, on it

**Comment by fromanirh** [comment]
@davidvossel this PR has no unmerged dependencies, once approved it could be merged anytime.

**Pull request merged**